src/openms/source/QC/Ms2IdentificationRate.cpp

```diff
@@ -0,0 +1,159 @@
+// --------------------------------------------------------------------------
+//                   OpenMS -- Open-Source Mass Spectrometry
+// --------------------------------------------------------------------------
+// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,
+// ETH Zurich, and Freie Universitaet Berlin 2002-2018.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Chris Bielow$
+// $Authors: Patricia Scheil, Swenja Wagner$
+// --------------------------------------------------------------------------
+
+//#include <OpenMS/QC/Ms2IdentificationRate.h>
+#include <include/OpenMS/QC/Ms2IdentificationRate.h>
+#include <algorithm>
+#include <iostream>
+
+
+
+namespace OpenMS
+{
+
+  Ms2IdentificationRate::Ms2IdentificationRate() = default;
+  Ms2IdentificationRate::~Ms2IdentificationRate() = default;
+
+  Int64 Ms2IdentificationRate::countPeptideId_(std::vector<PeptideIdentification> peptide_id, bool force_fdr)
+  {
+    Int64 counter{};
+
+    counter += count_if(peptide_id.begin(), peptide_id.end(), [force_fdr](PeptideIdentification const & x)
+    {
+
+      if ( !x.getHits().empty() )
```

Comment:
for readibility:
instead of
```
if (!a)
{
  if (!b) ...
  else ...
}
else throw ...
```

you could use:
```
if (a) throw ()

if (b) ...
```

produces less indentation and is easier to grasp